hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/multitenant/TestMultiTenantGateKeeperRangerPlugin.java

```diff
@@ -0,0 +1,221 @@
+  private List<String> policyIdsCreated = new ArrayList<String>();
+
+  private static OzoneConfiguration conf;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    conf = new OzoneConfiguration();
+    simulateOzoneSiteXmlConfig();
+  }
+
+  @AfterClass
+  public static void shutdown() {
+  }
+
+  private static void simulateOzoneSiteXmlConfig() {
+    conf.setStrings(OZONE_RANGER_HTTPS_ADDRESS_KEY, RANGER_ENDPOINT);
+    conf.setStrings(OZONE_OM_RANGER_HTTPS_ADMIN_API_USER, RANGER_ENDPOINT_USER);
+    conf.setStrings(OZONE_OM_RANGER_HTTPS_ADMIN_API_PASSWD,
+        RANGER_ENDPOINT_USER_PASSWD);
+  }
+
+  @Test
+  public void testMultiTenantGateKeeperRangerPlugin() throws Exception {
+    simulateOzoneSiteXmlConfig();
+    MultiTenantGateKeeper omm = new MultiTenantGateKeeperRangerPlugin();
+    omm.init(conf);
+
+    try {
+      OzoneMultiTenantPrincipal group1Principal = getTestPrincipal("tenant1",
+          "groupTestAdmin", GROUP_PRINCIPAL);
+      OzoneMultiTenantPrincipal group2Principal = getTestPrincipal("tenant1",
+          "groupTestUsers", GROUP_PRINCIPAL);
+      groupIdsCreated.add(omm.createGroup(group1Principal));
+      groupIdsCreated.add(omm.createGroup(group2Principal));
+
+      OzoneMultiTenantPrincipal userPrincipal =
+          getTestPrincipal("tenant1", "user1Test", USER_PRINCIPAL);
+      usersIdsCreated.add(omm.createUser(userPrincipal, groupIdsCreated));
+
+      AccessPolicy tenant1VolumeAccessPolicy = createVolumeAccessPolicy(
+          "vol1", "tenant1", "Users");
+      policyIdsCreated.add(omm.createAccessPolicy(tenant1VolumeAccessPolicy));
+
+      AccessPolicy tenant1BucketCreatePolicy = allowCreateBucketPolicy(
+          "vol1", "tenant1", "Users");
+      policyIdsCreated.add(omm.createAccessPolicy(tenant1BucketCreatePolicy));
+
+      AccessPolicy tenant1BucketAccessPolicy = allowAccessBucketPolicy(
+          "vol1", "tenant1", "Users", "bucket1");
+      policyIdsCreated.add(omm.createAccessPolicy(tenant1BucketAccessPolicy));
+
+      AccessPolicy tenant1KeyAccessPolicy = allowAccessKeyPolicy(
+          "vol1", "tenant1", "Users", "bucket1");
+      policyIdsCreated.add(omm.createAccessPolicy(tenant1KeyAccessPolicy));
+
+    } catch (Exception e) {
+      Assert.fail(e.getMessage());
+    } finally {
+      for (String id : policyIdsCreated) {
+        omm.deletePolicy(id);
+      }
+      if (usersIdsCreated.size() == 1) {
+        omm.deleteUser(usersIdsCreated.get(0));
+      }
+      for (String id : groupIdsCreated) {
+        omm.deleteGroup(id);
+      }
+    }
+  }
+
+  OzoneMultiTenantPrincipal getTestPrincipal(String tenant, String id,
+      OzoneMultiTenantPrincipal.OzonePrincipalType type) {
+    OzoneMultiTenantPrincipal principal = new OzoneMultiTenantPrincipalImpl(
+        new BasicUserPrincipal(tenant),
+        new BasicUserPrincipal(id), type);
+    return principal;
+  }
+
+  AccessPolicy createVolumeAccessPolicy(String vol, String tenant,
```

Comment:
Nit. These may be moved to a class which catalogs the list of well known policies, and provides static reference to creating them.

Reply:
yup, We will refactor.